test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -3112,6 +3112,70 @@ def matcher_check_fn():
             compile_options={"dynamic": dynamic},
         )
 
+    @skipIfNoDynamoSupport
+    @skipIfNoONEDNN
+    # TODO: investigate options of torch.compile in fbcode
+    @unittest.skipIf(IS_FBCODE, "Failing in fbcode")
+    @parametrize("has_bias", [True, False])
+    @parametrize("dtype", [torch.float, torch.bfloat16])
+    @parametrize("dynamic", [True, False])
+    def test_da8w8_sym_act_sym_wgt_with_int_mm(self, has_bias, dtype, dynamic):
+        r"""
+        This testcase check if we can match the int8_dynamic_activation_int8_weight int8 linear pattern from torchao,
+        when activation is symmetrically quantized dynamically & weights are symmetrically quantized (statically)
+        The pattern is:
+            (no bias) _int_mm -> convert_element_type -> (expand_a) -> mul -> mul
+        or
+            (with bias) pattern_no_bias -> add
+        Expansion of the scale of activation is optional.
+        """
+        if dtype == torch.bfloat16 and not torch.ops.mkldnn._is_mkldnn_bf16_supported():
+            return
+        M = 32
+        in_feature = 32
+        out_feature = 64
+        q_min, q_max = -32, 31
+
+        class Mod(torch.nn.Module):
+            def __init__(self, dtype: torch.dtype, has_bias: bool):
+                super().__init__()
+                self.dtype = dtype
+                self.has_bias = has_bias
+                self.b = torch.randint(
+                    q_min, q_max, [in_feature, out_feature], dtype=torch.int8
+                )
+                self.a_scale = torch.rand([M, 1], dtype=dtype) * 0.01 + 0.01
+                self.b_scale = torch.rand([out_feature]) * 0.01 + 0.01
+                self.b_scale = self.b_scale.to(dtype)
+                self.bias = torch.rand([out_feature], dtype=dtype) if has_bias else None
+
+            def forward(self, a):
+                a_reshaped = a.reshape(-1, a.size(-1))
+                c = torch._int_mm(a_reshaped, self.b)
+                c = c.to(self.dtype)
+                a_scale = self.a_scale.expand(c.shape)
+                c = c * a_scale
+                c = c * self.b_scale
+                if self.has_bias:
+                    c = c + self.bias
+                return c
```

Comment:
It seems we reshape input from 3D into 2D and never reshape it back. Is it the same of the implementation in TorchAO?

Reply:
It's 2D & stays 2D

Reply:
From this added UT:
```
a = torch.randint(q_min, q_max, [1, M, in_feature], dtype=torch.int8)
```
Is this a 2D activation?

Reply:
No, I copy-pasted these UTs from @Xia-Weiwen's  base PR. I think the motivation is to have reshapes in the graph for pattern-matching.

Reply:
Just confirmed that the UTs still pass even if input shape is 2D. Not sure what the motivation for 3D shape was, then

Reply:
Sorry, my changes were not pushed earlier with ghstack. They have been pushed now.